#!/bin/bash
HOME=/panfs/panasas01/sscm/qh18484
scripts=$HOME/bin/eczema_gwas_fu/bayesian_fm/paintor
PAINTOR=$HOME/bin/PAINTOR_V3.0
PAINTOR_ANALYSIS=$HOME/analysis/bayesian_fm/paintor/1k/euro
analysis=$HOME/analysis/bayesian_fm/RefPanel/1kGenomes
gwas=/panfs/panasas01/sscm/qh18484/data/gwas/paternoster2015
utils=$HOME/bin/eczema_gwas_fu/utils
#Submit test Paintor run
cd $PAINTOR_ANALYSIS

function paintor_mcmc 
{
snp=$1
interval=$2
chrom=$(grep $snp $gwas/paternoster_2015_index_snps_sorted.txt | cut -f2)
cd $PAINTOR_ANALYSIS
mkdir -p chr${chrom}.${snp}/${interval}.mcmc
cd chr${chrom}.${snp}/${interval}.mcmc
ln -s $analysis/chr${chrom}.${snp}.${interval}.locus2.processed ./chr${chrom}.${snp}.${interval}
ln -s $analysis/chr${chrom}.${snp}.${interval}.locus2.ld ./chr${chrom}.${snp}.${interval}.ld
echo "chr${chrom}.${snp}.${interval}" >chr${chrom}.${snp}.${interval}.txt

#Create dummy annotations.
no_snps=$(wc -l chr${chrom}.${snp}.${interval} | cut -d " " -f1)
echo "dummy" >chr${chrom}.${snp}.${interval}.annotations
for (( c=1; c<=${no_snps}; c++)) ; do echo "0" ; done >>chr${chrom}.${snp}.${interval}.annotations
qsub -v input=chr${chrom}.${snp}.${interval}.txt,max_causal=5 $scripts/sub_paintor_mcmc.sh
}


function paintor_exact
{
snp=$1
interval=$2
chrom=$(grep $snp $gwas/paternoster_2015_index_snps_sorted.txt | cut -f2)
cd $PAINTOR_ANALYSIS
mkdir -p chr${chrom}.${snp}/${interval}.exact
cd chr${chrom}.${snp}/${interval}.exact
ln -s $analysis/chr${chrom}.${snp}.${interval}.locus2.processed ./chr${chrom}.${snp}.${interval}
ln -s $analysis/chr${chrom}.${snp}.${interval}.locus2.ld ./chr${chrom}.${snp}.${interval}.ld
echo "chr${chrom}.${snp}.${interval}" >chr${chrom}.${snp}.${interval}.txt

#Create dummy annotations.
no_snps=$(wc -l chr${chrom}.${snp}.${interval} | cut -d " " -f1)
echo "dummy" >chr${chrom}.${snp}.${interval}.annotations
for (( c=1; c<=${no_snps}; c++)) ; do echo "0" ; done >>chr${chrom}.${snp}.${interval}.annotations
qsub -v input=chr${chrom}.${snp}.${interval}.txt,max_causal=2 $scripts/sub_paintor_exact.sh
}


function paintor_mcmc_haploblock 
{
snp=$1
interval=$2
chrom=$(grep $snp $gwas/paternoster_2015_index_snps_sorted.txt | cut -f2)
cd $PAINTOR_ANALYSIS
mkdir -p chr${chrom}.${snp}/${interval}.mcmc
cd chr${chrom}.${snp}/${interval}.mcmc
ln -s $analysis/${snp}.${interval}.processed ./chr${chrom}.${snp}.${interval}
ln -s $analysis/${snp}.${interval}.ld ./chr${chrom}.${snp}.${interval}.ld
echo "chr${chrom}.${snp}.${interval}" >chr${chrom}.${snp}.${interval}.txt

#Create dummy annotations.
no_snps=$(wc -l chr${chrom}.${snp}.${interval} | cut -d " " -f1)
echo "dummy" >chr${chrom}.${snp}.${interval}.annotations
for (( c=1; c<=${no_snps}; c++)) ; do echo "0" ; done >>chr${chrom}.${snp}.${interval}.annotations
qsub -v input=chr${chrom}.${snp}.${interval}.txt,max_causal=5 $scripts/sub_paintor_mcmc.sh
}


function paintor_exact_haploblock
{
snp=$1
interval=$2
chrom=$(grep $snp $gwas/paternoster_2015_index_snps_sorted.txt | cut -f2)
cd $PAINTOR_ANALYSIS
mkdir -p chr${chrom}.${snp}/${interval}.exact
cd chr${chrom}.${snp}/${interval}.exact
ln -s $analysis/${snp}.${interval}.processed ./chr${chrom}.${snp}.${interval}
ln -s $analysis/${snp}.${interval}.ld ./chr${chrom}.${snp}.${interval}.ld
echo "chr${chrom}.${snp}.${interval}" >chr${chrom}.${snp}.${interval}.txt

#Create dummy annotations.
no_snps=$(wc -l chr${chrom}.${snp}.${interval} | cut -d " " -f1)
echo "dummy" >chr${chrom}.${snp}.${interval}.annotations
for (( c=1; c<=${no_snps}; c++)) ; do echo "0" ; done >>chr${chrom}.${snp}.${interval}.annotations
qsub -v input=chr${chrom}.${snp}.${interval}.txt,max_causal=2 $scripts/sub_paintor_exact.sh
}

for snp in rs61813875 rs10791824 rs6419573 rs12188917 rs2212434 rs2918307 rs4809219 rs4713555 rs2041733 rs2944542 rs145809981 rs6827756 rs12730935 rs4312054 rs1249910 rs2592555 rs7127307 rs2038255 rs6602364 rs6473227 rs7512552 rs112111458 rs7625909 rs4643526 rs10199605 rs10214237 rs12951971 rs1057258 rs6872156 rs7016497 rs2905493 rs1799986 rs2227483 rs7146581 rs11657987 rs77714197 rs3917265 rs13152362 rs4705962 rs2218565 rs7512552 rs12730935 rs12295535 rs11923593 rs2143950 rs17881320 rs41293864
do
for interval in 5000 50000 250000 500000 1500000 
do
paintor_exact $snp $interval
paintor_mcmc $snp $interval
done
done

#Analysis using intervals generated by gpart.
for snp in rs61813875 rs10791824 rs6419573 rs12188917 rs2212434 rs2918307 rs4809219 rs4713555 rs2041733 rs2944542 rs145809981 rs6827756 rs12730935 rs4312054 rs1249910 rs2592555 rs7127307 rs2038255 rs6602364 rs6473227 rs7512552 rs112111458 rs7625909 rs4643526 rs10199605 rs10214237 rs12951971 rs1057258 rs6872156 rs7016497 rs2905493 rs1799986 rs2227483 rs7146581 rs11657987 rs77714197 rs3917265 rs13152362 rs4705962 rs2218565 rs7512552 rs12730935 rs12295535 rs11923593 rs2143950 rs17881320 rs41293864
do
for interval in gpart bigld
do
paintor_exact_haploblock $snp $interval
paintor_mcmc_haploblock $snp $interval
done
done
